agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -3507,6 +3508,27 @@ bool backhaul_manager::get_media_type(const std::string &interface_name,
     return result;
 }
 
+std::unique_ptr<link_metrics_collector>
+backhaul_manager::create_link_metrics_collector(const sLinkInterface &link_interface) const
+{
+    ieee1905_1::eMediaType media_type = link_interface.media_type;
+    ieee1905_1::eMediaTypeGroup media_type_group =
+        static_cast<ieee1905_1::eMediaTypeGroup>(media_type >> 8);
```

Comment:
I believe this should work as well, and is [what CppCoreGuidelines advises](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Res-construct):
```suggestion
    ieee1905_1::eMediaTypeGroup media_type_group {media_type >> 8};
```

For consistency, the other initializer should probably use `{}` instead of `=` as well, then.

Reply:
Done

Reply:
Sorry, I cannot do that since I get a compilation error:
```
../agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp:3401:61: error: invalid conversion from 'int' to 'ieee1905_1::eMediaTypeGroup' [-fpermissive]
     ieee1905_1::eMediaTypeGroup media_type_group{media_type >> 8};
                                                  ~~~~~~~~~~~^~~~
```

Since a cast is required anyway, I'll leave it as it was